motion/motion.go

```diff
@@ -52,23 +57,46 @@ func NewMotionDetector(args MotionConfig) *motionDetector {
 }
 
 type motionDetector struct {
-	flooredFrames FrameLoop
-	diffFrames    FrameLoop
-	firstDiff     bool
-	useOneDiff    bool
-	tempThresh    uint16
-	deltaThresh   uint16
-	countThresh   int
-	warmerOnly    bool
-	start         int
-	rowStop       int
-	columnStop    int
-	count         int
-
-	debug *debugTracker
+	flooredFrames     FrameLoop
+	diffFrames        FrameLoop
+	firstDiff         bool
+	dynamicThresh     bool
+	useOneDiff        bool
+	tempThresh        uint16
+	defaultTempThresh uint16
+	deltaThresh       uint16
+	countThresh       int
+	warmerOnly        bool
+	start             int
+	rowStop           int
+	columnStop        int
+	count             int
+	background        [lepton3.FrameRows][lepton3.FrameCols]uint16
+	backgroundWeight  float32
+	debug             *debugTracker
+	previewFrames     int
+}
+
+func (d *motionDetector) calculateThreshold(backAverage float64) {
+	oldTemp := d.tempThresh
+	d.tempThresh = uint16(math.Min(backAverage, float64(d.defaultTempThresh)))
+	if oldTemp != d.tempThresh {
+		d.debug.update("thresh", int(d.tempThresh))
+	}
 }
 
 func (d *motionDetector) Detect(frame *lepton3.Frame) bool {
+	if d.dynamicThresh {
```

Comment:
I think you should avoid adjusting the threshold when processing frames that were affected by FFC. See the isAffectedByFFC function in this file.